File: app/src/main/java/com/example/megabest/app/features/favourite/FavouriteMovieViewModel.java
package com.example.megabest.app.features.favourite;

import android.app.Application;
import android.util.Log;

import androidx.annotation.NonNull;
import androidx.lifecycle.AndroidViewModel;
import androidx.lifecycle.LiveData;

import com.example.megabest.domain.dataSource.MovieLocalDataSource;
import com.example.megabest.domain.repositories.MovieLocalRepository;
import com.example.megabest.entities.FavouriteMovie;
import com.example.megabest.domain.repositories.FavouriteMovieRepository;

import java.util.List;

import static android.content.ContentValues.TAG;

public class FavouriteMovieViewModel extends AndroidViewModel {
    private FavouriteMovieRepository moviesRepository;
    private MovieLocalRepository movieLocalRepository ;

    public FavouriteMovieViewModel(@NonNull Application application) {
        super(application);
        moviesRepository= FavouriteMovieRepository.getInstance(application);
        MovieLocalDataSource movieLocalDataSource = MovieLocalDataSource.getInstance(application);

    }


    public void insertFavouriteMovie(FavouriteMovie movie){
        moviesRepository.insertFavouriteMovie(movie);
    }

    public void deleteFavouriteMovie(FavouriteMovie movie){
        moviesRepository.deleteFavouriteMovie(movie);
    }
    public LiveData<List<FavouriteMovie>> getFavouriteMovies(){
//        Log.d(TAG, "getFavouriteMovies: "+movieLocalRepository.getFavouriteMovies().getValue().size());
        return moviesRepository.getFavouriteMovies();
    }

}

File: app/src/main/java/com/example/megabest/app/features/favourites/FavouriteMovieViewModel.kt
package com.example.megabest.app.features.favourites

import android.app.Application
import androidx.lifecycle.AndroidViewModel
import androidx.lifecycle.LiveData
import androidx.lifecycle.MutableLiveData
import com.example.megabest.domain.repositories.MoviesRepository
import com.example.megabest.domain.repositories.MoviesRepositoryImpl
import com.example.megabest.domain.usecases.SearchMoviesUseCase
import com.example.megabest.entities.FavouriteMovie
import com.example.megabest.entities.Movie

class FavouriteMovieViewModel(
    application: Application,
    private val searchMoviesUseCase: SearchMoviesUseCase = SearchMoviesUseCase()
) : AndroidViewModel(application) {

    fun insertFavouriteMovie(movie: FavouriteMovie?) {
        searchMoviesUseCase.insertFavouriteMovie(movie)
    }

    fun deleteFavouriteMovie(movie: FavouriteMovie?) {
        searchMoviesUseCase.deleteFavouriteMovie(movie)
    }

    val favouriteMovies: LiveData<List<FavouriteMovie>>
        get() = searchMoviesUseCase.getFavouriteMovies()
}